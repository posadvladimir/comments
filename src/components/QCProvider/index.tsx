
import { QueryClient, QueryClientProvider } from 'react-query'

import type { TChildrenProps } from 'src/ts/types/children';

const queryClient = new QueryClient();

const QCProvider: React.FC<TChildrenProps> = ({ children }) => (
    <QueryClientProvider client={queryClient}>
        {children}
    </QueryClientProvider>
);

export default QCProvider;