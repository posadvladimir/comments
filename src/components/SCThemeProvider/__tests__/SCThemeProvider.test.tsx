
import { render, screen } from '@testing-library/react';

import { runSnapshotTest } from 'src/common/tests/snapshot';

import QCProvider from '..';

describe('QCProvider', () => {
    const childComponentTestText = 'Test Child';

    const component = (
        <QCProvider>
            <div>{childComponentTestText}</div>
        </QCProvider>
    );

    test('отображаются дочерние компоненты', () => {
        render(component);

        expect(screen.getByText(childComponentTestText)).toBeInTheDocument();
    });

    runSnapshotTest(component);
});