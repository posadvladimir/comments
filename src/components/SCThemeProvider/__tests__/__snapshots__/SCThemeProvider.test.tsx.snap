// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QCProvider соотвествует снепшоту 1`] = `
<div>
  Test Child
</div>
`;

exports[`SCThemeProvider соотвествует снепшоту 1`] = `
<div>
  Test Child
</div>
`;
